modules/Movecraft/src/main/java/net/countercraft/movecraft/craft/BaseCraft.java

```diff
@@ -91,7 +93,60 @@ public BaseCraft(@NotNull CraftType type, @NotNull World world) {
         materials = new Counter<>();
         audience = Audience.empty();
     }
+    //Start of Adding Craft Datatag System
+    public void setDataTag(String key, Object data) {
+      this.dataTags.put(key, data);
+    }
+    public void removeDataTag(String key) {
+      this.dataTags.remove(key);
+    }
+    public Object getDataTag(String key) {
+      return this.dataTags.getOrDefault(key, "null");
+    }
+    public Collection<Object> getAllTagValues() {
+      return this.dataTags.values();
+    }
+    public boolean hasDataKey(String key) {
+      return this.dataTags.containsKey(key);
+    }
+    public boolean hasDataValue(Object value) {
+      return this.dataTags.containsValue(value);
+    }    
+    //End of Added Craft Datatag System
+    
+    //Start of Simple Mechanism for getting Blocks
+    public Set<Block> getBlockData(BlockData bkd){
+       Set<Block> blocks = new HashSet<>();
+       for (MovecraftLocation mloc : this.getHitBox()){
+          if (mloc.toBukkit(this.getWorld()).getBlock().getBlockData() == bkd) {
+             blocks.add(mloc.toBukkit(this.getWorld()).getBlock());
 
+          }
+       }
+       return blocks;
+    }
+    public Set<Block> getBlockName(String s){
+       Set<Block> blocks = new HashSet<>();
+
+       for (MovecraftLocation l : (this.getHitBox())){
+          if (l.toBukkit(this.getWorld()).getBlock().getType().toString().toLowerCase().contains(s.toLowerCase())) {
+             blocks.add(l.toBukkit(this.getWorld()).getBlock());
+
+          }
+       }
+       return blocks;
+    }
+    public Set<Block> getBlockType(Material mat){
+       Set<Block> blocks = new HashSet<>();
+       for (MovecraftLocation mloc : this.getHitBox()){
+          if (mloc.toBukkit(this.getWorld()).getBlock().getType() == mat) {
+             blocks.add(mloc.toBukkit(this.getWorld()).getBlock());
+
+          }
+       }
+       return blocks;
+    }
+    //End of Simple Mechanism for getting Blocks
```

Comment:
This is just a wrapping for one method to interface with the HitBox, and implies a certain amount of optimization.  I'd prefer if we leave it as is.